src/middleware/bearer-auth/index.ts

```diff
@@ -0,0 +1,56 @@
+import type { Context } from '../../context'
+import type { Next } from '../../hono'
+import { timingSafeEqual } from '../../utils/buffer'
+
+const TOKEN_REGEXP = /^Bearer ([A-Za-z0-9._~+/-]+=*) *$/
```

Comment:
I'm building a Discord bot with Hono, and Discord API has an Auth header in the form of <code>Authorization: Bot &#x2329;<i>token</i>&#x232A;</code> as well as the usual `Bearer` version ([reference](https://discord.com/developers/docs/reference#authentication-example-bot-token-authorization-header)). I'd be happy if this middleware supports this kind of headers out-of-the-box. Possibly via `options.prefix` parameter (defaults to `Bearer` unless explicitly specified) in `bearerAuth`? I'm not sure how many these irregular headers other than Discord API exist in the wild though.

Reply:
Hi @yudai-nkt 

I didn't know `Authorization: Bot <token>`. Good suggestion! I've added the `prefix` option.
Thank you!

Reply:
LGTM, thanks!

> I didn't know `Authorization: Bot <token>`. 

Me neither. Discord is the only example I know. 🤣